posthog/hogql/query.py

```diff
@@ -74,23 +75,30 @@ def execute_hogql_query(
     clickhouse_context = HogQLContext(
         team_id=team.pk, enable_select_queries=True, person_on_events_mode=team.person_on_events_mode
     )
-    clickhouse = print_ast(
+    clickhouse_sql = print_ast(
         select_query, context=clickhouse_context, dialect="clickhouse", settings=settings or HogQLSettings()
     )
 
-    results, types = insight_sync_execute(
-        clickhouse,
+    tag_queries(
+        team_id=team.pk,
+        query_type=query_type,
+        has_joins="JOIN" in clickhouse_sql,
+        has_json_operations="JSONExtract" in clickhouse_sql or "JSONHas" in clickhouse_sql,
```

Comment:
How accurate do we want to be with these two tags? Since we have the full modified pre-printer AST available to us (lazy tables expanded, etc), we can just use a quick visitor that checks if there's a select query with any joins for a 100% accurate answer. See e.g. `AggregationFinder` in `property.py`. The current simple check will also catch ARRAY JOIN-s, which I'm not sure if it's the intention or not.

As for JSONExtract, that's trickier to autodetect based on HogQL, as property access is expanded in the printer. So yeah, this here is the safest way though there might be some extremely rare duplicates.

All in all, do with the context above what you think is best :).